Implemented the Discord Rich Presense support

Wohlstand opened the pull request:
Recently, I made the fully functional Discord Rich Presense that can be enabled on supported platforms at the `settings/thextech.ini` file by adding the `discord-rpc = 1` value into the `[gameplay]` section.

Fixes #39

ds-sloth commented:
Looks exciting! Should be a nice and simple add-on but useful for players. Do you want a full code review or is it simple enough to just merge?

Wohlstand commented:
Mostly it simple, but still needs a review of parts (mainly the the Integrator module I added). Also, the "Integrator" module I made, also ca be used to implement integrations with external speed-run counters like LiveSplit and something also.

Wohlstand commented:
Forgot to say, I fixed the next:
- I made sure that it will call the `Discord_RunCallbacks()` as documentation requires.
- I made this feature being disabled by default, and users who wants to enable it, should change the `discord-rpc` field to `1` at the `[gameplay]` section in order to enable this thing.
- Fixed the condition to enable Discord RPC support at 32-bit Windows builds (they still can work on 64bit OS and still can interact with 64-bit Discord client).
